aws/resource_aws_sagemaker_notebook_instance_test.go

```diff
@@ -136,6 +136,68 @@ func TestAccAWSSagemakerNotebookInstance_disappears(t *testing.T) {
 	})
 }
 
+func TestAccAWSSagemakerNotebookInstance_directInternetAccess_disabled(t *testing.T) {
+	var notebook sagemaker.DescribeNotebookInstanceOutput
+	notebookName := resource.PrefixedUniqueId(sagemakerTestAccSagemakerNotebookInstanceResourceNamePrefix)
+	var resourceName = "aws_sagemaker_notebook_instance.foo"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSagemakerNotebookInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSSagemakerNotebookInstanceConfigDirectInternetAccess(notebookName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSagemakerNotebookInstanceExists(resourceName, &notebook),
+					testAccCheckAWSSagemakerNotebookInstanceName(&notebook, notebookName),
+					testAccCheckAWSSagemakerNotebookInstanceDirectInternetAccess(&notebook, "Disabled"),
+
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_notebook_instance.foo", "name", notebookName),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_notebook_instance.foo", "direct_internet_access", "Disabled"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerNotebookInstance_directInternetAccess_enabled(t *testing.T) {
```

Comment:
It looks like this acceptance test is failing while the others pass:

```
--- FAIL: TestAccAWSSagemakerNotebookInstance_directInternetAccess_enabled (255.12s)
    testing.go:538: Step 0 error: Check failed: Check 3/5 error: DirectInternetAccess not configured correctly: want Enabled got Disabled
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (259.02s)
--- PASS: TestAccAWSSagemakerNotebookInstance_directInternetAccess_disabled (292.10s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (296.42s)
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (314.73s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (443.07s)
```

Did you mean to point it to the same `direct_internet_access = "Disabled"` configuration? The easiest fix might be to parameterize that argument for the configuration:

```go
func testAccAWSSagemakerNotebookInstanceConfigDirectInternetAccess(notebookName, directInternetAccess string) string {
	return fmt.Sprintf(`
resource "aws_vpc" "main" {
  cidr_block = "10.0.0.0/16"
}
resource "aws_subnet" "main" {
  vpc_id     = "${aws_vpc.main.id}"
  cidr_block = "10.0.1.0/24"
  tags = {
    Name = "tf-acc-test-sagemaker-notebook-instance-direct-internet-access"
  }
}
resource "aws_security_group" "foo" {
  name        = "foo"
  vpc_id      = "${aws_vpc.main.id}"
  description = "foo bar baz"
  tags = {
    Name = "foo_sg"
  }
}
resource "aws_sagemaker_notebook_instance" "foo" {
  name                   = %[1]q
  role_arn               = "${aws_iam_role.foo.arn}"
  instance_type          = "ml.t2.medium"
  subnet_id              = "${aws_subnet.main.id}"
  security_groups        = ["${aws_security_group.foo.id}"]
  direct_internet_access = %[2]q
}
resource "aws_iam_role" "foo" {
  name               = %[1]q
  path               = "/"
  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
}
data "aws_iam_policy_document" "assume_role" {
  statement {
    actions = ["sts:AssumeRole"]
    principals {
      type        = "Service"
      identifiers = ["sagemaker.amazonaws.com"]
    }
  }
}
`, notebookName, directInternetAccess)
}
```

Then updating the two `Config` calls:

```go
// in TestAccAWSSagemakerNotebookInstance_directInternetAccess_disabled
Config: testAccAWSSagemakerNotebookInstanceConfigDirectInternetAccess(notebookName, "Disabled"),

// in TestAccAWSSagemakerNotebookInstance_directInternetAccess_enabled
Config: testAccAWSSagemakerNotebookInstanceConfigDirectInternetAccess(notebookName, "Enabled"),
```

Reply:
I'll refactor this